packages/basti/src/aws/elasticache/get-elasticache-memcached-clusters.ts

```diff
@@ -0,0 +1,75 @@
+import { DescribeCacheClustersCommand } from '@aws-sdk/client-elasticache';
+
+import {
+  parseMemcachedCacheClusteWithSpecificNoderResponse,
+  parseMemcachedCacheClusterResponse,
+} from './parse-elasticache-memcached-response.js';
+import { elasticacheClient } from './elasticache-client.js';
+
+import type { CacheCluster } from '@aws-sdk/client-elasticache';
+import type { AwsElasticacheMemcachedCluster } from './elasticache-types.js';
+
+export interface getCacheClusterInput {
+  identifier: string;
+}
+
+export async function getMemcachedClusters(): Promise<
+  AwsElasticacheMemcachedCluster[]
+> {
+  const memcachedClusters = await getRawMemcachedClusters();
+  return memcachedClusters.map(cacheCluster => {
+    return parseMemcachedCacheClusterResponse(cacheCluster);
+  });
+}
+
+export async function getRawMemcachedClusters(): Promise<CacheCluster[]> {
+  const { CacheClusters } = await elasticacheClient.send(
+    new DescribeCacheClustersCommand({
+      ShowCacheNodeInfo: true,
+      ShowCacheClustersNotInReplicationGroups: true,
+    })
+  );
+
+  if (!CacheClusters) {
+    throw new Error(`Invalid response from AWS.`);
+  }
+  const memcachedClusters = CacheClusters.filter(
+    cacheCluster => cacheCluster.Engine === 'memcached'
+  );
+  return memcachedClusters;
+}
+
+export async function getMemcachedCluster(
+  identifier: string
+): Promise<AwsElasticacheMemcachedCluster> {
+  const { CacheClusters } = await elasticacheClient.send(
+    new DescribeCacheClustersCommand({
+      ShowCacheNodeInfo: true,
+      ShowCacheClustersNotInReplicationGroups: true,
+      CacheClusterId: identifier,
+    })
+  );
+  if (!CacheClusters) {
+    throw new Error(`Invalid response from AWS.`);
+  }
+  return parseMemcachedCacheClusterResponse(CacheClusters[0]);
+}
+
+export async function getMemcachedNode(
+  identifier: string
+): Promise<AwsElasticacheMemcachedCluster> {
+  const cacheClusters = await getRawMemcachedClusters();
+  const cluster = cacheClusters.find(cluster =>
```

Comment:
Is it possible to fetch only the specific cluster we need using some filters in the `DescribeCacheClustersCommand`? This would prevent possible issues when the user has a large number of clusters in their account.

Reply:
In this case we get Memcached Node's id and we try to get it's information.
Sadly , in elasticache SDK there isn't a function allowing us to search for the node directly , we need to find the cluster containing the node and parse it from there